application/src/main/kotlin/net/corda/v5/application/messaging/FlowMessaging.kt

```diff
@@ -8,12 +10,47 @@ import net.corda.v5.base.types.MemberX500Name
 interface FlowMessaging {
 
     /**
-     * Creates a communication session with [party]. Subsequently you may send/receive using this session object. Note
-     * that this function does not communicate in itself, the counter-flow will be kicked off by the first send/receive.
+     * Creates a communication session with another member. Subsequently you may send/receive using this session object.
+     * Note that this function does not communicate in itself, the counter-flow will be kicked off by the first
+     * send/receive.
+     *
+     * Initiated flows are initiated with context based on the context of the initiating flow at the point in time this
+     * method is called. The context of the initiating flow is snapshotted by the returned session. Altering the flow
+     * context has no effect on the context of the session after this point, and therefore it has no effect on the
+     * context of the initiated flow either.
+     *
+     * @param x500Name The X500 name of the member to communicate with.
+     *
+     * @return The session.
      */
     @Suspendable
     fun initiateFlow(x500Name: MemberX500Name): FlowSession
 
+    /**
+     * Creates a communication session with another member. Subsequently you may send/receive using this session object.
+     * Note that this function does not communicate in itself, the counter-flow will be kicked off by the first
+     * send/receive.
+     *
+     * This overload takes a builder of context properties. Any properties set or modified against the context passed to
+     * this builder will be propagated to initiated flows and all that flow's initiated flows and sub flows down the
+     * stack. The properties passed to the builder are pre-populated with the current flow context properties, see
+     * [FlowContextProperties]. Altering the current flow context has no effect on the context of the session after the
+     * builder is applied and the session returned by this method, and therefore it has no effect on the context of the
+     * initiated flow either.
+     *
+     * @param x500Name The X500 name of the member to communicate with.
+     * @param flowContextPropertiesBuilder A builder of context properties.
+     *
+     * @return The session.
+     *
+     * @throws IllegalArgumentException if the builder tries to set a property for which a platform property already
+     * exists or if the key is prefixed by [CORDA_RESERVED_PREFIX]. See also [FlowContextProperties]. Any other
+     * exception thrown by the builder will also be thrown through here and should be avoided in the provided
+     * implementation, see [FlowContextPropertiesBuilder].
+     */
+    @Suspendable
+    fun initiateFlow(x500Name: MemberX500Name, flowContextPropertiesBuilder: FlowContextPropertiesBuilder): FlowSession
```

Comment:
Since we're going through and putting examples in some of the Kdocs, I think that this might be a good candidate for an example.

The rest of the kdocs look nice though.

Reply:
yeah, we discussed already and I forgot, will add